src/EditorFeatures/CSharpTest/ConvertToInterpolatedString/ConvertConcatenationToInterpolatedStringTests.cs

```diff
@@ -322,5 +322,40 @@ void M()
     }
 }");
         }
+
+        [WorkItem(16820, "https://github.com/dotnet/roslyn/issues/16820")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertToInterpolatedString)]
+        public async Task TestWithMultipleStringConcatinations()
+        {
+            await TestAsync(
+@"public class C
+{
+    void M()
+    {
+        var v = ""A"" + 1 + [||]""B"" + ""C"";
```

Comment:
test with more strange stuff.  like "A" + "B" + 1   and    "A" + 1 + "B" + "C" + "D"